packages/base-contract/src/index.ts

```diff
@@ -80,7 +80,7 @@ export class BaseContract {
             // Awaiting https://github.com/Microsoft/TypeScript/pull/13288 to be merged
         } as any;
         if (_.isUndefined(txDataWithDefaults.gas) && !_.isUndefined(estimateGasAsync)) {
-            txDataWithDefaults.gas = await estimateGasAsync(txData);
+            txDataWithDefaults.gas = await estimateGasAsync(txDataWithDefaults as any);
```

Comment:
We need to be passing in `from` to `estimateGasAsync`. @LogvinovLeon, do you see any issues with this change?

Reply:
This is actually not true, we only need `from` when sending transactions, not when estimating gas. Reverted this change.